src/index.ts

```diff
@@ -15,29 +16,47 @@ const services = {
   }),
   log: new Logger({
     stdout: process.stdout,
-    stderr: createWriteStream("./error.log"),
+    stderr: createWriteStream("./error.log", { flags: "a" }),
   }),
   scheduler: new Scheduler(client),
 };
 
-services.store.onError((err) => services.log.error(err));
-
 const eventStream = createMessageStream(
   config,
   client,
   createCommandMatcher(config, commands, services)
 );
 
-client.once("ready", () => services.log.info("CatJammer is ready"));
-
 const eventSubscription = eventStream.subscribe({
   next: handleCommand,
 });
 
-client.login(config.token).catch((err) => services.log.error(err));
-
-process.on("SIGINT", () => {
+const cleanup = (msg = "CatJammer closing cleanly") => {
   eventSubscription.unsubscribe();
   client.destroy();
-  services.log.info("CatJammer closing cleanly");
+  services.log.info(msg);
+};
+
+const onError = (err: unknown) => {
+  services.log.error("!!FATAL!!", err);
+  cleanup("CatJammer is closing unexpectedly");
+  process.exit(1);
+};
+
+services.store.onError(onError);
```

Comment:
Should we close on storage errors? If persistence isn't working for any reason, I think it is reasonable to prevent operation.

Reply:
I'm not sure. On one hand, the bot is still functional without data being saved between boots and shouldn't effect anything else plus it would still be reported in logs that it is failing. 

On the other hand, if the bot exits on these errors, it should cause the issues to be dealt with straight away rather then once it reboots and significant amounts of data might have been lost.

Probably safer to quit on error.